### DISCUSSION TITLE
Creating a 3D Wing with Pyansys Geometry

### DISCUSSION THREAD

**Discussion by BharathTatavarthi** [ansys/pyansys-geometry#983] [General]
I am currently working on a project focused on optimizing the aerodynamic shape of a fixed-wing UAV, and I am using the NACA 2412 airfoil as the reference profile. I am encountering difficulties in creating a 3D wing geometry in PyANSYS Geometry using SpaceClaim Mode. Initially, I attempted to work with a 2D airfoil profile using the PyVista spline command and encountered errors when trying to extrude it.

I am seeking guidance on whether it is possible to create a 3D wing in PyANSYS Geometry with varying chord sections. If it is feasible, could you assist me in understanding how to efficiently create a 3D wing with different chord sections?
my team member @santoshdoddipatla is also asked about this

**Comment by RobPasMue** [comment]
Hi @Bharath158 - thanks for opening this discussion. I will close it since it is a duplicate of #984. Currently, splines are not supported on PyAnsys Geometry on the ``Sketch`` API and thus they cannot be extruded. This depends on features that are not implemented yet on our backend. If there are any updates I will ping you both in #984. Sorry for the inconveniences!

In the meantime you can approximate your spline using straight lines or curves and incrementing the number of points in your spline. I know it is not ideal but it might be a short-term solution.
